ert3/evaluator/_evaluator.py

```diff
@@ -1,57 +1,136 @@
+    ) -> None:
+        self._phase_name = phase_name
+        if not 0 <= phase <= self._phase_count:
+            raise ValueError(
+                "Phase must be an integer between (inclusive) 0 and {self._phase_count}"
+            )
+
+        if phase == 0:
+            self._job_start_time = int(time.time())
+
+        if phase == self._phase_count:
+            self._job_stop_time = int(time.time())
+
+        self._phase = phase
+
+    def isIndeterminate(self) -> bool:
+        return False
+
+    def setIndeterminate(self, indeterminate: Union[bool, None]) -> None:
+        raise NotImplementedError("This function is not in use in ert3")
+
+
 def _run(
     ensemble_evaluator: EnsembleEvaluator,
+    run_model: DummyBaseRunModel,
 ) -> Dict[int, Dict[str, ert.data.RecordTransmitter]]:
     result: Dict[int, Dict[str, ert.data.RecordTransmitter]] = {}
     with ensemble_evaluator.run() as monitor:
-        realization_ids = set()
-        realizations_completed = set()
+        run_model.setPhase(0, "Running simulations...", indeterminate=False)
         for event in monitor.track():
             if isinstance(event.data, dict) and event.data.get("status") in [
                 ENSEMBLE_STATE_STOPPED,
                 ENSEMBLE_STATE_FAILED,
             ]:
                 monitor.signal_done()
                 if event.data.get("status") == ENSEMBLE_STATE_FAILED:
+                    run_model._failed = True
+                    run_model._fail_message = "Ensemble evaluation failed"
                     raise RuntimeError("Ensemble evaluation failed")
             if event["type"] == EVTYPE_EE_TERMINATED and isinstance(event.data, bytes):
+                run_model.setPhase(1, "Simulations completed.")
                 result = pickle.loads(event.data)
-            if isinstance(event.data, dict) and "reals" in event.data:
-                for real_id in event.data["reals"]:
-                    realization_ids.add(real_id)
-                    real_status = event.data["reals"][real_id].get("status")
-                    if (
-                        real_status == REALIZATION_STATE_FINISHED
-                        and real_id not in realizations_completed
-                    ):
-                        realizations_completed.add(real_id)
-                        print(
-                            f"Realization {real_id} completed successfully"
-                            f" ({len(realizations_completed)}/{len(realization_ids)})"
-                        )
 
     return result
 
 
 def evaluate(
-    ensemble: Ensemble,
+    ensemble: Ensemble, cli: bool = True
 ) -> Dict[int, Dict[str, ert.data.RecordTransmitter]]:
     config = EvaluatorServerConfig()
+
+    run_model = DummyBaseRunModel(1)
+    if cli:
+        tracker = EvaluatorTracker(
+            run_model,
+            config.host,
+            config.port,
+            1,
+            0,
+            token=config.token,
+            cert=config.cert,
```

Comment:
This is the `EvaluatorTracker.__init__` signature:

```python
        model: BaseRunModel,
        host: str,
        port: str,
        token: str = None,
        cert: str = None,
        next_ensemble_evaluator_wait_time: int = 5,
```

Seems the intervals was left in there, we can probably just remove them.